bundles/commonalities/tools.vitruv.applications.cbs.commonalities/src/tools/vitruv/applications/cbs/commonalities/CbsCommonalitiesApplication.java

```diff
@@ -0,0 +1,23 @@
+package tools.vitruv.applications.cbs.commonalities;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import tools.vitruv.framework.applications.AbstractVitruvApplication;
+import tools.vitruv.framework.change.processing.ChangePropagationSpecification;
+
+public class CbsCommonalitiesApplication extends AbstractVitruvApplication {
+
+	@Override
+	public Set<ChangePropagationSpecification> getChangePropagationSpecifications() {
+		Set<ChangePropagationSpecification> specs = new HashSet<ChangePropagationSpecification>();
+		//specs.add(new UmlToJavaChangePropagationSpecification()); // TODO
+		//specs.add(new JavaToUmlChangePropagationSpecification());
```

Comment:
Should they be commented in?

Reply:
The CbsCommonalitiesApplication is not yet used or supported. The CBSExecutionTestCompiler used for the test cases searches for generated change propagation specifications classes to provide a combined change propagation. Ideally the Commonalities language should probably provide/generate a combined change propagation specification since it is closest to knowing which change propagation specifications it generates via the Reactions language. But this is not the case currently. Once this is done, this combined specification can then be referenced here and then these commonality-based change propagation specifications can be used as a Vitruv application, outside of test cases.
I added a code comment for now.

Reply:
Alternatively: As soon as the Commonalities language is shipped as part of Vitruv, and the code for the commonalities gets generated in the original source project (without having to launch another Eclipse application instance which has the Commonalities language installed), the generated change propagation specifications could probably be referenced in this application class directly.

Reply:
Okay, perfect. Currently the code generation for Commonalities within the Maven build is not possible. I added an issue in the framework project for that. We have to wait until that problem is fixed to enable code generation in the CBS Commonalities project and then enable the `CbsCommonalitiesApplication`.